assets/js/src/settings/pages/send_with/send_with.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import classnames from 'classnames';
+import ReactStringReplace from 'react-string-replace';
+
+import { useSelector } from 'settings/store/hooks';
+import { MssStatus } from 'settings/store/types';
+import { t } from 'common/functions';
+import { Link } from 'react-router-dom';
+
+export default function SendWith() {
+  const isNewUser = useSelector('isNewUser')();
+  const isMssActive = useSelector('isMssActive')();
+  const { mssStatus } = useSelector('getKeyActivationState')();
+  const isMssKeyValid = mssStatus !== null && mssStatus !== MssStatus.INVALID;
+  const freePlanUrl = (window as any).mailpoet_free_plan_url;
+
+  return (
+    <ul className="mailpoet_sending_methods">
+      <li
+        data-group="mailpoet"
+        className={classnames({
+          mailpoet_active: isMssActive,
+          mailpoet_invalid_key: !isMssKeyValid,
+        })}
+      >
+        <div className="mailpoet_sending_method_description">
+          <h3>{t('mssTitle')}</h3>
+          <div className="mailpoet_description">
+            {isMssActive && (
+              <span id="mailpoet_sending_method_active_text">
+                <strong>{t('youreSendingWithMss')}</strong>
+              </span>
+            )}
+            {!isMssActive
+              && (
+                <span id="mailpoet_sending_method_inactive_text">
+                  <strong>{t('solveSendingProblems')}</strong>
+                </span>
+              )}
+            <ul className="sending-method-benefits mailpoet_success">
+              <li className="mailpoet_success_item">{t('mssBenefit1')}</li>
+              <li className="mailpoet_success_item">{t('mssBenefit2')}</li>
+              <li className="mailpoet_success_item">{t('mssBenefit3')}</li>
+              <li className="mailpoet_success_item">{t('mssBenefit4')}</li>
+              <li className="mailpoet_success_item">{t('mssBenefit5')}</li>
+            </ul>
+            <p className="mailpoet_sending_methods_help">
+              <a
+                target="_blank"
+                rel="noopener noreferrer"
+                data-beacon-article="5a3d4c260428631938003802"
+                href="https://kb.mailpoet.com/article/235-video-guide-sending-options"
+                className={classnames('mailpoet_badge mailpoet_badge_video', {
+                  mailpoet_badge_video_grey: !isNewUser,
+                })}
+              >
+                <span className="dashicons dashicons-format-video" />
+                {t('seeVideo')}
+              </a>
+            </p>
+          </div>
+        </div>
+        <div className={classnames('mailpoet_status', {
+          mailpoet_invalid_key: !isMssActive || !isMssKeyValid,
```

Comment:
I think this should be `isMssActive && !isMssKeyValid`. The div  should have `mailpoet_invalid_key` class only when MSS is activated and the key is invalid. 

I also noticed a difference in CSS.
This is how `mailpoet_invalid_key` renders in old settings: https://d.pr/i/1lI9EF
This is new settings: https://d.pr/i/FQyw63

Reply:
I agree that this condition looks wrong, but it's the equivalent of what we had before
https://github.com/mailpoet/mailpoet/blob/3.46.8/views/settings/mta.html#L94
```
<div class="mailpoet_status <% if(settings.mta_group != 'mailpoet' or not mss_key_valid) %> mailpoet_invalid_key <% endif %>">
```
`settings.mta_group != 'mailpoet'` is `!isMssActive` and `not mss_key_valid` is `!isMssKeyValid`.

For the difference in CSS, I think you are mixing the case when MSS is active vs when another method is active.
When MSS is active and the key is invalid, it should have the red border. When another method is active and the key is invalid, the border is grey.

Reply:
Ahh. I see now. It works as expected. Sorry. The condition really confused me.